repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/services/datastore/RemoteDatastoreServiceTest.groovy

```diff
@@ -21,142 +21,315 @@ package features.services.datastore
+
+        return [reposeValveLauncher, reposeLogSearch]
     }
 
     static def waitUntilReadyToServiceRequests(ReposeLogSearch reposeLogSearch) {
-        def clock = new SystemClock()
-        try {
-            waitForCondition(clock, '35s', '1s') {
-                return (reposeLogSearch.awaitByString("Repose ready", 1, 35, TimeUnit.SECONDS).size() > 0)
-            }
-        } catch (TimeoutException ignored) {
-            return false
-        }
+        reposeLogSearch.awaitByString("Repose ready", 1, 60, TimeUnit.SECONDS)
     }
 
     def "When a limit has not been reached, request should pass"() {
-        given: "the rate-limit has not been reached"
+        given: "three Repose instances are started, two to handle traffic, one to act as the remote datastore"
+        (repose1, repose1LogSearch) = startRepose('repose1', repose1Port, targetPort, datastorePort, true)
+        (repose2, repose2LogSearch) = startRepose('repose2', repose2Port, targetPort, datastorePort, true)
+        (remoteDatastore, reposeRemoteLogSearch) =
+            startRepose('remote', reposeRemotePort, targetPort, datastorePort, false)
+
+        and: "they are ready to service requests"
+        waitUntilReadyToServiceRequests(repose1LogSearch)
+        waitUntilReadyToServiceRequests(repose2LogSearch)
+        waitUntilReadyToServiceRequests(reposeRemoteLogSearch)
+
+        and: "the rate-limit has not been reached"
         def headers = ["X-PP-User": "user", "X-PP-Groups": "group"]
 
-        def messageChain1
-        def messageChain2
-        when: "the user sends their request"
-        5.times {
-            messageChain1 = deproxy.makeRequest(url: reposeOneEndpoint, headers: headers)
-            messageChain2 = deproxy.makeRequest(url: reposeTwoEndpoint, headers: headers)
-
-            then: "the request is not rate-limited, and passes to the origin service"
-            assert messageChain1.receivedResponse.code as Integer == SC_OK
-            assert messageChain1.handlings.size() == 1
-            assert messageChain2.receivedResponse.code as Integer == SC_OK
-            assert messageChain2.handlings.size() == 1
-        }
-
-        and: "the user sends their request after the rate-limit has been reached"
-        messageChain1 = deproxy.makeRequest(url: reposeOneEndpoint, headers: headers)
-        messageChain2 = deproxy.makeRequest(url: reposeTwoEndpoint, headers: headers)
-
-        then: "the request is rate-limited, and passes to the origin service"
+        when: "the user sends 10 requests total (5 to each Repose instance)"
+        def manyMessageChains1 = (1..5).collect { deproxy.makeRequest(url: repose1Endpoint, headers: headers) }
+        def manyMessageChains2 = (1..5).collect { deproxy.makeRequest(url: repose2Endpoint, headers: headers) }
+
+        then: "the requests are not rate-limited and pass to the origin service"
+        manyMessageChains1.every { it.receivedResponse.code as Integer == SC_OK }
```

Comment:
Oh derp, i was reading it with the semantics of for each, but every checks a condition against every element.